tracing-observation/src/main/java/com/couchbase/client/tracing/observation/CouchbaseSenderContext.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023 Couchbase, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.couchbase.client.tracing.observation;
+
+import io.micrometer.observation.transport.Kind;
+import io.micrometer.observation.transport.SenderContext;
+
+import static com.couchbase.client.core.util.Validators.notNull;
+
+/**
+ * A {@link SenderContext} for Couchbase.
+ *
+ * @author Marcin Grzejszczak
+ */
+public class CouchbaseSenderContext extends SenderContext<Object> {
```

Comment:
Could you provide more detail on what this is up to, as I couldn't find much online on a quick search for SenderContext. 

Is it responsible for sending traceparent from https://www.w3.org/TR/trace-context/?

Reply:
Ok thanks.  That's pretty much what I'd guessed, just wanted to confirm.  (And TIL about the b3 form of trace propagation)